packages/web-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,194 @@
+  public disabled?: boolean;
+
+  /**
+   * size
+   * Used to set the size of all the tab controls, which effects text size and margins. Three sizes: small, medium and large.
+   */
+  @attr
+  public size?: TabsSize = 'medium';
+
+  /**
+   * reserve-selected-tab-space
+   * Controls whether or not to make the text of the active tab bold. Bold text will cause a slight movement in the ui of the tabs. Defaults to false.
+   */
+  @attr({ attribute: 'reserve-selected-tab-space', mode: 'boolean' })
+  public reserveSelectedTabSpace?: boolean;
+
+  /**
+   * calculateAnimationProperties
+   *
+   * Recalculates the active tab offset and scale.
+   * These values will be applied to css variables that control the tab active indicator position animations
+   */
+  private calculateAnimationProperties(tab: Tab) {
+    this.activeTabOffset = this.getSelectedTabPosition(tab);
+    this.activeTabScale = this.getSelectedTabScale(tab);
+  }
+
+  /**
+   * getSelectedTabPosition - gets the x or y coordinates of the tab
+   */
+  private getSelectedTabPosition(tab: Tab): number {
+    if (this.orientation === 'horizontal') {
+      return this.previousActiveTabData.x - (tab.getBoundingClientRect().x - this.getBoundingClientRect().x);
+    } else return this.previousActiveTabData.y - (tab.getBoundingClientRect().y - this.getBoundingClientRect().y);
+  }
+
+  /**
+   * getSelectedTabScale - gets the scale of the
+   */
+  private getSelectedTabScale(tab: Tab): number {
+    if (this.orientation === 'horizontal') {
+      return this.previousActiveTabData.width / tab.getBoundingClientRect().width;
+    } else return this.previousActiveTabData.height / tab.getBoundingClientRect().height;
+  }
+
+  /**
+   * clearAnimationProperties
+   * resets animation values to defaults. removes the animated css class
+   */
+  private clearAnimationProperties(tab: Tab) {
+    this.previousActiveTabData = { id: '', x: 0, y: 0, height: 0, width: 0 };
+    this.activeTabOffset = 0;
+    this.activeTabScale = 1;
+    tab.classList.remove('animated');
+  }
+
+  /**
+   * animationLoop
+   * runs through all the operations required for setting the tab active indicator to its starting location, ending location, and applying the animated css class to the tab.
+   * @param tab
+   */
+  private animationLoop(tab: Tab) {
+    // If selected, set the animation properties.
+    if (tab.ariaSelected && tab.ariaSelected === 'true') {
+      // get the offset for active indicator
+      // the offset at the start of the animation will be at the location of the previously selected tab
+      this.calculateAnimationProperties(tab);
+      // set the active indicator offset and scale of the active indicator to css vars
+      this.setTabScaleCSSVar();
+      this.setTabOffsetCSSVar();
+
+      // clear properties for next animation
+      this.clearAnimationProperties(tab);
+    }
+
+    this.previousActiveTabData = this.activeTabData;
+    this.calculateAnimationProperties(tab);
```

Comment:
What are we calculating if the tab itself is not selected?

Reply:
The tab is selected here. Here we have to calculate the ending location of the animation. But, probably these two statements should be moved below where activeTabOffset and scale are checked

Reply:
I think I need to play with this a bit to understand the loop that's happening here.

Reply:
Geoff Cox referred me to this article - https://css-tricks.com/animating-layouts-with-the-flip-technique/ You may want to check it out.